pkg/config/remote/uptane/partial_client_test.go

```diff
@@ -149,9 +151,164 @@ func TestPartialClientVerifyInvalidRotation(t *testing.T) {
 	testRepository2 := newTestRepository(2, nil, targets1, []*pbgo.File{{Path: "datadog/2/APM_SAMPLING/id/1", Raw: target1content}})
 
 	err = client.Update(testRepository2.toPartialUpdate())
+	// TODO in the whole file: use "assert.ErrorAs" with specific error type
 	assert.Error(t, err)
 	_, err = client.Targets()
 	assert.Error(t, err)
 	_, err = client.TargetFile("datadog/2/APM_SAMPLING/id/1")
 	assert.Error(t, err)
 }
+
+func TestPartialClientRootKeyRotation(t *testing.T) {
+	_, targetFileMeta := generateTarget()
+	directorTargetMetadata := data.TargetFiles{
+		"datadog/2/APM_SAMPLING/id/1": targetFileMeta,
+	}
+
+	repository1 := newTestRepository(1, nil, directorTargetMetadata, nil)
+	config.Datadog.Set("remote_configuration.director_root", repository1.directorRoot)
+	config.Datadog.Set("remote_configuration.config_root", repository1.configRoot)
+
+	client, err := NewPartialClient()
+	require.NoError(t, err)
```

Comment:
if you use this more than once I would suggest to do

```
assert := assert.New(t)
```

and 
```
assert.NoError(err)
```

Reply:
Done in 85e9be0